presto-geospatial/src/main/java/com/facebook/presto/plugin/geospatial/GeoFunctions.java

```diff
@@ -1586,7 +1592,218 @@ public static Double stSphericalLength(@SqlType(SPHERICAL_GEOGRAPHY_TYPE_NAME) S
         return sum * 1000;
     }
 
+    @SqlNullable
+    @Description("Returns TRUE if and only if no points of right lie in the exterior of left, and at least one point of the interior of left lies in the interior of right")
+    @ScalarFunction("ST_Contains")
+    @SqlType(BOOLEAN)
+    public static Boolean stSphericalContains(@SqlType(SPHERICAL_GEOGRAPHY_TYPE_NAME) Slice left, @SqlType(SPHERICAL_GEOGRAPHY_TYPE_NAME) Slice right)
+    {
+        OGCGeometry leftGeometry = deserialize(left);
+        if (leftGeometry.isEmpty()) {
+            return null;
+        }
+
+        OGCGeometry rightGeometry = deserialize(right);
+        if (rightGeometry.isEmpty()) {
+            return null;
+        }
+
+        validateSphericalType("ST_Contains", leftGeometry, EnumSet.of(POLYGON));
+        validateSphericalType("ST_Contains", rightGeometry, EnumSet.of(POINT));
+
+        Polygon polygon = (Polygon) leftGeometry.getEsriGeometry();
+        Point point = (Point) rightGeometry.getEsriGeometry();
+
+        if (point.getY() == MAX_LATITUDE || point.getY() == MIN_LATITUDE) {
+            throw new PrestoException(INVALID_FUNCTION_ARGUMENT, "When applied to SphericalGeography inputs, ST_Contains only supports points which are not poles.");
+        }
+
+        if (containsPole(polygon)) {
+            throw new PrestoException(INVALID_FUNCTION_ARGUMENT, "When applied to SphericalGeography inputs, ST_Contains only supports polygons which do not enclose poles.");
+        }
+
+        return getNumberIntersectionsWithPolygon(polygon, point) % 2 == 1;
```

Comment:
the name of this method appears strange because points don't intersect polygons; I suggest to rename this method to `contains`, change its return type to `boolean` and fold `% 2 == 1` logic into it.

```
    return contains(polygon, point);

    private static boolean contains(Polygon polygon, Point point)
    {
        ...
        return intersectionSum % 2 == 1;
    }
```